src/common/remoteExtension.ts

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import {ExtensionMessage, MessagingChannel} from "./extensionMessaging";
+import {InterProcessMessageSender} from "./interProcessMessageSender";
+
+export class RemoteExtension {
```

Comment:
The code that use this class should be independent on how we talk with the extension. You want to say extension.startPackager() you don't really care if you send it through a socket, or if tomorrow we change it and we put it in the same process for some reason.
I also like putting all the "untyped" code in a single place, so we can just use it blindly in any other places (It's easy to send the wrong arguments to .sendMessage() because of the any[] type), here it's at least in a single place, which I think will make it easier to make sure that all the calls are correct.